backend/app/Http/Controllers/CategoryController.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Category;
+use Illuminate\Http\Request;
+
+class CategoryController extends Controller
+{
+    public function index()
+	{
+		$categories = Category::all();
+		return $categories;
+	}
+    public function getCategoryDetails($categoryId)
+	{
+		return Category::findOrFail($categoryId);
+	}
+
+	public function updateCategoryDetails(Request $request, $categoryId)
+	{
+		$category = Category::where('id', $categoryId)->first();
+	
+		$request->validate([
+			'title' => 'required|string',
+			'description'=>'required|string'
+		]);
+	
+		Category::whereId($categoryId)->update([
+			'title'=>$request->title,
+			'description'=>$request->description
+		]);
+	
+		return response()->json([
+			'message' => 'Successfully Updated!',
+		], 200);
+	}
+
+    public function categoryQuestions()
+	{
+		$categories = Category::with(['questions', 'questions.choices', 'questions.choices.answer'])->get();
```

Comment:
too long, apply next line
```suggestion
		$categories = Category::with([
		    'questions', 
		    'questions.choices', 
		    'questions.choices.answer'
		])->get();
```